--- rp-pppoe-3.10/src/Makefile.in.ldflags~	2012-05-24 13:31:24.545364342 +0200
+++ rp-pppoe-3.10/src/Makefile.in	2012-05-24 13:32:05.033858167 +0200
@@ -67,16 +67,16 @@ all: $(TARGETS)
 	@echo "Type 'make install' as root to install the software."
 
 pppoe-sniff: pppoe-sniff.o if.o common.o debug.o
-	@CC@ -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o
+	@CC@ $(LDFLAGS) -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o
 
 pppoe-server: pppoe-server.o if.o debug.o common.o md5.o libevent/libevent.a @PPPOE_SERVER_DEPS@
-	@CC@ -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent
+	@CC@ $(LDFLAGS) -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent
 
 pppoe: pppoe.o if.o debug.o common.o ppp.o discovery.o
-	@CC@ -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o
+	@CC@ $(LDFLAGS) -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o
 
 pppoe-relay: relay.o if.o debug.o common.o
-	@CC@ -o pppoe-relay relay.o if.o debug.o common.o
+	@CC@ $(LDFLAGS) -o pppoe-relay relay.o if.o debug.o common.o
 
 pppoe.o: pppoe.c pppoe.h
 	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o pppoe.o pppoe.c
@@ -113,25 +113,25 @@ relay.o: relay.c relay.h pppoe.h
 
 # Linux-specific plugin
 rp-pppoe.so: plugin/libplugin.a plugin/plugin.o
-	@CC@ -o rp-pppoe.so -shared plugin/plugin.o plugin/libplugin.a
+	@CC@ $(LDFLAGS) -Wl,--warn-unresolved-symbols -o rp-pppoe.so -shared plugin/plugin.o plugin/libplugin.a
 
 plugin/plugin.o: plugin.c
-	@CC@ '-DRP_VERSION="$(VERSION)"' $(CFLAGS) -I$(PPPD_INCDIR) -c -o plugin/plugin.o -fPIC plugin.c
+	@CC@ '-DRP_VERSION="$(VERSION)"' $(CFLAGS) -I$(PPPD_INCDIR) -c -o plugin/plugin.o -fno-PIE -fPIC plugin.c
 
 plugin/libplugin.a: plugin/discovery.o plugin/if.o plugin/common.o plugin/debug.o
 	ar -rc $@ $^
 
 plugin/discovery.o: discovery.c
-	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/discovery.o -fPIC discovery.c
+	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/discovery.o -fno-PIE -fPIC discovery.c
 
 plugin/if.o: if.c
-	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/if.o -fPIC if.c
+	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/if.o -fno-PIE -fPIC if.c
 
 plugin/debug.o: debug.c
-	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/debug.o -fPIC debug.c
+	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/debug.o -fno-PIE -fPIC debug.c
 
 plugin/common.o: common.c
-	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/common.o -fPIC common.c
+	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o plugin/common.o -fno-PIE -fPIC common.c
 
 install: all
 	-mkdir -p $(DESTDIR)$(sbindir)
