--- rp-pppoe-3.10/scripts/pppoe-init.in.orig	2012-05-11 10:34:48.447455209 +0100
+++ rp-pppoe-3.10/scripts/pppoe-init.in	2012-05-11 10:36:02.490233760 +0100
@@ -10,6 +10,15 @@
 # Copyright (C) 2000 Roaring Penguin Software Inc.  This software may
 # be distributed under the terms of the GNU General Public License, version
 # 2 or any later version.
+#
+### BEGIN INIT INFO
+# Provides: pppoe
+# Required-Start: $network
+# Required-Stop: $network
+# Default-Start: 2 3 4 5
+# Short-Description: Manage PPPoE Connections
+# Description: Manage PPPoE Connections
+### END INIT INFO
 
 # Source function library if it exists
 test -r /etc/rc.d/init.d/functions && . /etc/rc.d/init.d/functions
@@ -49,7 +58,7 @@
         echo ""
         ;;
 
-    restart)
+    restart|reload)
 	$0 stop
 	$0 start
 	;;
@@ -59,7 +68,7 @@
 	;;
 
     *)
-        echo "Usage: pppoe {start|stop|restart|status}"
+        echo "Usage: pppoe {start|stop|restart|reload|status}"
         exit 1
 esac
 
