diff -up rp-pppoe-4.0/src/Makefile.in.2~ rp-pppoe-4.0/src/Makefile.in
--- rp-pppoe-4.0/src/Makefile.in.2~	2023-04-30 15:53:12.346103054 +0200
+++ rp-pppoe-4.0/src/Makefile.in	2023-04-30 15:55:56.020784231 +0200
@@ -118,25 +118,25 @@ relay.o: relay.c relay.h pppoe.h
 
 # Linux-specific plugin
 rp-pppoe.so: plugin/plugin.o plugin/libplugin.a
-	@CC@ -o $@ -shared $^ $(LDFLAGS)
+	@CC@ -o $@ -shared $^ $(LDFLAGS) -Wl,--warn-unresolved-symbols
 
 plugin/plugin.o: plugin.c
-	@CC@ -DPLUGIN=1 '-DRP_VERSION="$(RP_VERSION)"' $(CFLAGS) -I$(PPPD_INCDIR) -c -o $@ -fPIC $<
+	@CC@ -DPLUGIN=1 '-DRP_VERSION="$(RP_VERSION)"' $(CFLAGS) -I$(PPPD_INCDIR) -c -o $@ -fno-PIE -fPIC $<
 
 plugin/libplugin.a: plugin/discovery.o plugin/if.o plugin/common.o plugin/debug.o
 	@AR@ -rc $@ $^
 
 plugin/discovery.o: discovery.c
-	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fPIC $<
+	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fno-PIE -fPIC $<
 
 plugin/if.o: if.c
-	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fPIC $<
+	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fno-PIE -fPIC $<
 
 plugin/debug.o: debug.c
-	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fPIC $<
+	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fno-PIE -fPIC $<
 
 plugin/common.o: common.c
-	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fPIC $<
+	@CC@ -DPLUGIN=1 $(CFLAGS) '-DRP_VERSION="$(RP_VERSION)"' -I$(PPPD_INCDIR) -c -o $@ -fno-PIE -fPIC $<
 
 install: all
 	-mkdir -p $(DESTDIR)$(sbindir)
